src/GovUk.Education.ExploreEducationStatistics.Content.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -1644,14 +1640,201 @@ private List<ReleaseFile> GenerateDataSetsForRelease(Release release, int number
+                .WithFile(_fixture.DefaultFile());
+
+            var client = BuildApp()
+                .AddContentDbTestData(context =>
+                {
+                    context.ReleaseFiles.Add(releaseFile);
+                })
+                .CreateClient();
+
+            var uri = $"/api/releases/{releaseFile.ReleaseId}/data-sets/{releaseFile.FileId}";
+
+            var response = await client.GetAsync(uri);
+
+            response.AssertNotFound();
+        }
+
+        [Fact]
+        public async Task AmendmentNotPublished_ReturnsNotFound()
+        {
+            Publication publication = _fixture.DefaultPublication()
+                .WithReleaseParents(
+                    _fixture.DefaultReleaseParent(publishedVersions: 2, draftVersion: true)
+                        .Generate(1))
+                .WithTopic(_fixture.DefaultTopic()
+                    .WithTheme(_fixture.DefaultTheme()));
+
+            File file = _fixture.DefaultFile();
+
+            ReleaseFile releaseFile0 = _fixture.DefaultReleaseFile()
+                .WithRelease(publication.Releases[0]) // the previous published version
+                .WithFile(file);
+
+            ReleaseFile releaseFile1 = _fixture.DefaultReleaseFile()
+                .WithRelease(publication.Releases[1]) // the latest published version
+                .WithFile(file);
+
+            ReleaseFile releaseFile2 = _fixture.DefaultReleaseFile()
+                .WithRelease(publication.Releases[2]) // the draft version
+                .WithFile(file);
+
+            var client = BuildApp()
+                .AddContentDbTestData(context =>
+                {
+                    context.ReleaseFiles.AddRange(releaseFile0, releaseFile1, releaseFile2);
+                })
+                .CreateClient();
+
+            var uri = $"/api/releases/{publication.Releases[2].Id}/data-sets/{releaseFile2.FileId}";
+
+            var response = await client.GetAsync(uri);
+
+            response.AssertNotFound();
+        }
+    }
+
+    private WebApplicationFactory<TestStartup> BuildApp(
+        ContentDbContext? contentDbContext = null)
     {
         return TestApp
             .ResetDbContexts()
             .ConfigureServices(services =>
             {
-                services.AddTransient(
-                    s => dataSetService ?? new DataSetService(s.GetRequiredService<ContentDbContext>()));
+                services.AddTransient<IReleaseRepository>(s => new ReleaseRepository(
+                    contentDbContext ?? s.GetRequiredService<ContentDbContext>()));
+                services.AddTransient<IDataSetService>(
+                    s =>
+                    {
+                        var context = contentDbContext ?? s.GetRequiredService<ContentDbContext>();
+
+                        return new DataSetService(
+                            context,
+                            new ReleaseRepository(context));
+                    });
```

Comment:
You could do this to simplify the configuration of the `IDataSetService` type:

```csharp
                services.AddTransient<IDataSetService>(
                    s => new DataSetService(
                        contentDbContext ?? s.GetRequiredService<ContentDbContext>(),
                        s.GetRequiredService<IReleaseRepository>()));
```

Alternatively you could leave that as it is, and remove the configuration for the `IReleaseRepository` type which isn't strictly needed with your current setup of `DataSetService`.